types/time.go

```diff
@@ -1628,6 +1628,12 @@ func parseSingleTimeValue(unit string, format string, strictCheck bool) (int64,
 		return 0, 0, 0, 0, ErrIncorrectDatetimeValue.GenWithStackByArgs(format)
 	}
 	riv := iv // Rounded integer value
+	if sign == int64(-1) {
+		iv, err = strconv.ParseInt(format[1:decimalPointPos], 10, 64)
+		if err != nil {
+			return 0, 0, 0, 0, ErrIncorrectDatetimeValue.GenWithStackByArgs(format)
+		}
+	}
```

Comment:
After another look, I think the problem occurs because of the incorrectness of `dv`.
Is it more reasonable to add a line `dv *= sign` to solve this problem?